### PR TITLE
JavaScript: Fix `InconsistentNew` performance regression.

### PR DESCRIPTION
**Pull request by xiemaisi**
As noted by Alex ET, this query has significantly regressed in performance from 1.18 on `rhino`. I can't immediately see any query or library changes that would have caused it, but it's not difficult to fix by refactoring the query; see first commit.

Looking at the results on `rhino`, I then noticed that the query was producing a humongous number of result links since it listed all calls to the function with `new` and all without, ballooning 17 alerts on `rhino` into >80K result tuples. The second commit fixes this by only showing one call with `new` and one without, picking the first one when sorted lexicographically by file name, line and column.

Comparing three versions of the query (before this PR, after the first commit, and after the second commit) shows no changes in alerts, and [reasonable performance](https://git.semmle.com/gist/max/9a4426944a04499c653d54a986debeb6) (internal link). Note that the table shows only the runtimes for the query itself, but I ran `definitions.ql` before to warm the cache. While some projects exhibit a slowdown of ten seconds or more, these are outliers, and worth the improved quality of results, I think.

Since this is a regression from 1.18, I'm proposing this as a hotfix.

### CODE REVIEW

File: javascript/ql/src/LanguageFeatures/InconsistentNew.ql [hunk 1]
```diff
@@ -95,8 +95,20 @@ predicate whitelistedCall(DataFlow::CallNode call) {
...
+  result = min(DataFlow::InvokeNode invk, string path, int line, int col |
+    f = getALikelyCallee(invk, isNew) and invk.hasLocationInfo(path, line, col, _, _) |
+    invk order by path, line, col
+  )
+}
+
 from Function f, DataFlow::NewNode new, DataFlow::CallNode call
-where f = getALikelyCallee(new, true) and
-      f = getALikelyCallee(call, false)
+where new = getFirstInvocation(f, true) and
```
Comment (esben-semmle) [inline javascript/ql/src/LanguageFeatures/InconsistentNew.ql]: Could we show the number of unlisted calls like: 

```
F is invoked as a constructor _here_ (and 17 other locations), and as a normal function _here_ (and 2 other locations) 
```
or
```
F is invoked as a constructor at 17 locations (such as _here_), and as a normal function at 2 locations (such as _here_)
```

I think being explicit in the query message about not showing all results will make the programmer happier than only alerting him once the query is run again.

Comment (xiemaisi) [reply javascript/ql/src/LanguageFeatures/InconsistentNew.ql]: I agree about making the query message more explicit, but adding the count is a non-trivial change. How about "F is sometimes invoked as a constructor (for example _here_) and sometimes as a normal function (for example _here_)"?

Comment (esben-semmle) [reply javascript/ql/src/LanguageFeatures/InconsistentNew.ql]: Agreed.

**Comment by esben-semmle** [comment]
This may not be tested with a dist-upgrade before the release, so ping @asger-semmle for a second review.

**Comment by xiemaisi** [comment]
> This might not be tested with a dist-upgrade before the release

We have one more distribution upgrade before the release (but of course more eyes are always good).

**Pull request merged**